assignments/02-collections-lifetimes/src/leet_code.rs

```diff
@@ -1,24 +1,79 @@
-use std::collections::HashMap;
+// use std::collections::HashMap;
 
-fn longest_equal_sequence_prescriptive(sequence) -> i32 {
-    todo!()
+fn longest_equal_sequence_prescriptive<T: PartialEq>(sequence: &Vec<T>) -> i32 {
+    if sequence.is_empty() {
+        return 0;
+    }
+
+    let mut max_length = 1;
+    let mut curr_length = 1;
+    let mut curr_value = &sequence[0];
+
+    for value in &sequence[1..] { 
+        if value == curr_value {
+            curr_length += 1;
+        } else {
+            if curr_length > max_length {
+                max_length = curr_length;
+            }
+            curr_length = 1;
+            curr_value = value; 
+        }
+    }
+
+    if curr_length > max_length {
+        max_length = curr_length;
+    }
+
+    max_length
 }
 
-fn longest_equal_sequence_functional(sequence) -> i32 {
-    todo!()
+fn longest_equal_sequence_functional<T: PartialEq + PartialOrd>(sequence: &Vec<T>) -> i32{
+    if sequence.is_empty() {
+        return 0;
+    }
+    let (max_streak, _) = sequence[0..sequence.len() - 1]
+        .iter()
+        .zip(sequence[1..sequence.len()].iter())
+        .map(|(first, second)| first < second)
+        .fold((1, 1), |(best, cur), equality| {
+            if equality {
+                (best.max(cur + 1), cur + 1)
+            } else {
+                (best, 1)
+            }
+        });
```

Comment:
You're super close - literally just using the wrong equality operator. In `map`, you want to check if `first` and `second` are _equal_